### PR TITLE
Implement proper error handling

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

File: include/xraylib-error.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ */
+typedef struct _xrl_error xrl_error;
+
+struct _xrl_error
+{
+  enum xrl_error_code code;
+  char *message;
+};
+
+xrl_error* xrl_error_new(enum xrl_error_code code, const char *format, ...) GNUC_PRINTF (2, 3);
```
Comment (kalvdans) [inline include/xraylib-error.h]: The code looks good, but I believe the error construction routines should not be exported from the library. Only the library should create errors and the user of the library should only need xrl_error_free and xrl_clear_error.

Comment (tschoonj) [reply include/xraylib-error.h]: Yep will move those to a private header and make sure they're not exported from the library (see #83 )

**Comment by kalvdans** [comment]
Looks good to me!

**Comment by tschoonj** [comment]
Hi @AndreWaehlisch 

I have updated the IDL bindings to use the new error handling API. I tested it quickly with IDL 7.1 for Linux but I would be grateful if you and/or your colleagues could give this a try to ensure everything still works...

Thanks in advance!

**Comment by AndreWaehlisch** [comment]
I am assuming bad input (e.g. Z=0) is supposed to fail silently now. In that case everything works fine for me with IDL8.5.1 (I am on openSuse). The following message becomes a little bit misleading however: `GETCOMPOUNDDATANISTBYNAME: Error: check preceding error messages`

Thanks for keeping the IDL binding alive :-)

**Comment by tschoonj** [comment]
Yep that's correct... bad input will no longer produce error messages except some special ones like `GETCOMPOUNDDATANISTBYNAME `. 

You're also correct about the misleading nature of those remaining error messages... Feel free to fix them 😄

**Pull request merged**